src/bp/lang-server/service/index.ts

```diff
@@ -0,0 +1,226 @@
+import { MLToolkit } from 'botpress/sdk'
+import { WrapErrorsWith } from 'errors'
+import fs from 'fs'
+import _ from 'lodash'
+import lru from 'lru-cache'
+import path from 'path'
+import { VError } from 'verror'
+
+import toolkit from '../../ml/toolkit'
+
+import { AvailableModel, LoadedBPEModel, LoadedFastTextModel, ModelFileInfo, ModelSet } from './typing'
+
+export default class LanguageService {
+  private _models: Dic<ModelSet> = {}
+  private _ready: boolean = false
+  private _cache
+
+  // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
+  private readonly FAST_TEXT_MODEL_REGEX = /(\w+)\.(\w+)\.(\d+)\.bin/i
+
+  // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
+  private readonly BPE_MODEL_REGEX = /(\w+)\.(\w+)\.bpe\.model/i
+
+  //This equals to 24H
+  private readonly _maxAgeCacheInMS = 86400000
+
+  constructor(public readonly dim: number, public readonly domain: string, private readonly langDir: string) {}
+
+  async initialize() {
+    if (Object.keys(this._models).length) {
+      throw new Error('Language Service already initialized')
+    }
+    this._cache = new lru({
+      maxAge: this._maxAgeCacheInMS
+    })
+
+    this._models = this._getModels()
+    this._logLoadingModels()
+    await Promise.all(Object.keys(this._models).map(this._loadModels.bind(this)))
+
+    this._ready = true
+  }
+
+  isReady(): boolean {
+    return this._ready
+  }
+
+  private _logLoadingModels = () => console.log(`Loading languages "${Object.keys(this._models).join(', ')}"`)
+  public listFastTextModels = (): AvailableModel[] => _.values(this._models).map(model => model.fastTextModel)
```

Comment:
If linting rules for chained methods we're not based on line length but instead on chaining methods it would be perfect;

 public listFastTextModels = (): AvailableModel[] => 
    _.values(this._models)
        .map(model => model.fastTextModel)

Reply:
this is removed in another PR